### DISCUSSION TITLE
Thoughts on Deta Space

### DISCUSSION THREAD

**Discussion by pietz** [deta/cloud-docs#575] [General]
I thought it would be nice to open a discussion about Deta Space. I've gotten access last week and while I haven't published anything, I read through the feature set and get the gist of it by now. A few thoughts and questions that come to my mind:

I find the general idea behind Deta Space super interesting and I can see the benefits that are described in the docs. However, there's also quite a few potential drawbacks. Yes, it's not easy today as a dev to focus on building the actual product but getting users onto the platform is arguably even harder. With Deta Space I will require my customers to create a Deta Space account (correct?). For someone who doesn't know Deta and the idea behind Space, it's kinda hard to explain why they need to create an account on Deta Space to use my service. It's just a little awkward to explain.

Is Deta Space the only planned way you want to make money? Because that feels like a huge bet on the general idea of Deta Space. I'm a little puzzled why you don't just offer a Pro tier for your 3 existing services. Micros would get a hardware upgrade, Drive would allow more storage. It just seems like a very easy way to make money given your current product. I mean, I would probably become a paying customer right away. It feels like you're leaving this source of income completely out of scope which I find puzzling.

Are you sure this will scale well? I love the serverless mindset and I see the benefits of what you're doing, but I find it hard to believe that the Deta Space concept would easily work for thousands of active users. You would essentially need thousands of active AWS lambda instances in the background. I believe that's what you're using under the hood.

Will I as a developer always have full control over my app? For small open source projects it's cool to view stuff like the sored data and structure of the database but there's tons of reasons why I, as the developer, wouldn't want that. For analytics and improvements reasons I might want to store most of the data centrally in one place. I know that this is possible with Deta Space but it also seems to go against the mindset of the platform. Is it guaranteed that users cannot see the code base since they're already looking at the database? I'm asking myself if should I better go elsewhere today if I don't want my users to be Deta users.

In a nutshell, while I find Deta Space very intriguing, I'm also sceptical if I want to use a cloud platform that my users actively need to become a part of. Today, I'm leaning towards "no". I would greatly appreciate if you could think about business models that revolve around what you currently offer: Well established cloud services with the easiest possible onboarding experience. To me, that's your biggest USP. This is also the reason I'm waiting for Deta Auth. Once I have an easy signup/signin service, I don't need much more from a cloud provider.

**Comment by abdelhai** [comment]
hi, @pietz great questions/critique!
i've written a lengthy response. after feedback from my colleagues, we think it should become a blog post since many other devs might find it useful/interesting. should be out this week. cheers!

**Reply by pietz** [reply]
sounds great! thank you for your effort!

**Reply by SlumberDemon** [reply]
Where can I find the blog post?

**Reply by xeust** [reply]
hi @pietz, this was my responsibility and I dropped the ball amongst other projects, sorry. 

We are getting this together and it should be out (on our new blog) shortly. Stay tuned.
